nemo/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -498,6 +525,9 @@ def _save_checkpoint(self, trainer: 'pytorch_lightning.Trainer', filepath: str)
             else:
                 finalize_fn()
 
+            if self.save_last_n_optim_states >= 0 and '-last' in filepath:
```

Comment:
I'm wondering (not sure) if this shouldn't be inside the `finalize_fn`? So called only after current checkpoint was actually finalized

Reply:
I think it makes sense to add it to the finalize function so that we always ensure we have at least `n` checkpoints with optimizer states. I made this change.